### ISSUE TITLE
Lua error when changing multiple enum settings in advanced settings

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#5834]
This bug seriously messes up the advanced settings menu, please add this to 0.4.16 milestone.

How to reproduce:

1. Use 210a339dce754a6cb03d4ef82f5bacc80b63262b
1. Clear `minetest.conf` to make sure we both use a common base
1. Start Minetest, open advanced settings and use it to change the settings in exactly this order:
1. Change `language` to “be”
1. Change `debug_log_level` to “verbose”
1. Change `language` a 2nd time to a different value, click “Save” and …

BOOM!

```
2017-05-27 22:08:35: ERROR[Main]: Main menu error: Runtime error from mod '' in callback handleMainMenuButtons(): ...netest/bin/../builtin/mainmenu/dlg_settings_advanced.lua:546: bad argument #2 to 'set' (string expected, got nil)
2017-05-27 22:08:35: ERROR[Main]: stack traceback:
2017-05-27 22:08:35: ERROR[Main]: 	[C]: in function 'set'
2017-05-27 22:08:35: ERROR[Main]: 	...netest/bin/../builtin/mainmenu/dlg_settings_advanced.lua:546: in function 'handle_buttons'
2017-05-27 22:08:35: ERROR[Main]: 	...zzy/src/minetest/git/minetest/bin/../builtin/fstk/ui.lua:148: in function 'handle_buttons'
2017-05-27 22:08:35: ERROR[Main]: 	...zzy/src/minetest/git/minetest/bin/../builtin/fstk/ui.lua:192: in function <...zzy/src/minetest/git/minetest/bin/../builtin/fstk/ui.lua:178>
```

Observations:
* It doesn't seem to matter which settings or values you choose, as long you use `enum` settings.
* Sometimes, after changing one of both settings, the displayed value of either setting assumes a value of the other setting 
* If you continue to change settings, above error may “randomly” pop up multiple times and survives restarts

**Comment by nerzhul** [comment]
A such critical setting bug should be fixed before release.

**Comment by pilino1234** [comment]
I can't reproduce the above crash, but am now getting quite different behaviour:
After clearing `minetest.conf` and then changing language to be in advanced settings menu, my config looks like this:

```language = be
maintab_LAST = settings
menu_last_game = minetest
```

Now, when I try to change the `debug_log_level`, the values in my config are set using the values from the language setting?! I.e. if I set `debug_log_level` to none, the first entry in the drop down list, my config actually displays the first value from the **language** list - `be`.

```
language = be
maintab_LAST = settings
menu_last_game = minetest
debug_log_level = be
```


Basically, it uses the language settings to set the debug_log_level settings...

**Comment by pilino1234** [comment]
It seems that MT will set any subsequent enum setting changes to a value from the first enum setting list that was changed during the session. If I change `debug_log_level` before `language`, `language` will be set to something like `verbose` after I change it.

**Comment by red-001** [comment]
this also happens with `bool`s

**Comment by red-001** [comment]
#5847 should fix this

**Comment by pilino1234** [comment]
Introduced by d5c3db9c4179e27a80c32ee95d6573ad0ec7137e
Will test #5847

**Comment by red-001** [comment]
The issue was that the C++ stored the drop down list along with the dropdown menu name in a vector and used to name to find the correct list. However when more then one drop down menu used the same name it would find the list for the one that was created first, instead of the latest one. This wouldn't be an issue if the code removed the dropdown list from the vector once the formspec element was removed but it didn't do this.
